src/components/ec-option-card/ec-option-card.vue

```diff
@@ -0,0 +1,199 @@
+<template>
+  <component
+    :is="componentTag"
+    v-bind="{
+      ...attrs,
+      ...componentProps,
+      'data-test': attrs['data-test'] ? `${attrs['data-test']} ec-option-card` : 'ec-option-card',
+    }"
+    class="ec-option-card"
+    :class="getOptionCardClasses()"
+    @click="setClickAction()"
+  >
+    <span
+      data-test="ec-option-card__title"
+    >
+      <ec-icon
+        v-if="optionCardIconName && optionTitle"
+        data-test="ec-option-card__title-img"
+        class="ec-option-card__title-img"
+        :class="getOptionIconClass()"
+        :name="optionCardIconName"
+        :alt="optionTitle"
+        :size="15"
+      />{{ optionTitle }}</span>
+    <p
+      v-if="optionCaption"
+      class="ec-option-card__caption"
+      :class="getOptionCaptionClass()"
+      data-test="ec-option-card__caption"
+    >{{ optionCaption }}</p>
+  </component>
+</template>
+
+<script setup lang="ts">
+import {
+  computed,
+  toRefs,
+  useAttrs,
+} from 'vue';
+
+import EcIcon from '../ec-icon';
+import type { IconName } from '../ec-icon/types';
+import { OptionCardType } from './types';
+
+const attrs = useAttrs();
+
+interface OptionCardProps {
+  isDisabled?: boolean,
+  isSubmit?: boolean,
+  optionTitle: string,
+  optionCaption?: string,
+  optionCardIconName?: IconName,
+  optionCardType?: OptionCardType,
```

Comment:
```suggestion
  title: string,
  caption?: string,
  iconName?: IconName,
  type?: OptionCardType,
```

Reply:
the only one I'm not sure about is`type?: OptionCardType` as there is also the icon type - and I feel is too confusing to just have a type here. I would prefer to be more specific to avoid confusion with `optionCardType?: OptionCardType`

Reply:
Sorry, I can't see icon type prop. BTW if we want to adjust icon, then we could introduce `iconProps?: IconProps` instead of different props `iconName` and `iconType`

Reply:
> _Sorry, I can't see icon type prop._ 

I meant in general and becuase we use an Icon here in the Option card - I didn't want the developer to get confused and think is related to the icon. 

```
// icon.vue

interface IconProps {
  name: IconName,
  size?: number,
  type?: IconType,
}
```

> _we could introduce iconProps?: IconProps instead of different props iconName and iconType_

Oh for this I think is fine because we only pass the icon name - we don't set an icon type here.

Reply:
I think it's fine to have just `type`, because we have no iconType here in this component. Moreover we see a type of that type prop - `OptionCardType`